File: server/src/main/com/broll/gainea/server/core/cards/impl/direct/C_PlaceSoldier.java
package com.broll.gainea.server.core.cards.impl.direct;

import com.broll.gainea.server.core.cards.DirectlyPlayedCard;
import com.broll.gainea.server.core.utils.PlayerUtils;

public class C_PlaceSoldier extends DirectlyPlayedCard {
    public C_PlaceSoldier() {
        //todo other picture
        super(7, "Verstärkung", "Jeder Spieler platziert einen Soldat");
    }

    @Override
    protected void play() {
        PlayerUtils.iteratePlayers(game, 1000, player -> {
            placeUnitHandler.placeSoldier(player);
        });
    }

}

File: server/src/main/com/broll/gainea/server/core/cards/impl/direct/C_PlaceSoldier.kt
package com.broll.gainea.server.core.cards.impl.direct

import com.broll.gainea.server.core.cards.DirectlyPlayedCard
import com.broll.gainea.server.core.utils.iteratePlayers

class C_PlaceSoldier :
    DirectlyPlayedCard(82, "Ritterdienst", "Jeder Spieler platziert einen Soldat") {
    override fun play() {
        game.iteratePlayers(1000) { player -> player.fraction.placeSoldier() }
    }
}
